pkg/products/rhssouser/reconciler_multitenant.go

```diff
@@ -0,0 +1,710 @@
+		r.Log.Error("Error getting all staging users", err)
+		return integreatlyv1alpha1.PhaseFailed, errors.Wrap(err, "Error getting all staging users")
+	}
+
+	newStagingUsers, err := r.reconcileStagingUsers(ctx, serverClient, mtUsers, existingStagingUsers, allRealms)
+	if err != nil {
+		r.Log.Error("Error getting all staging users", err)
+		return integreatlyv1alpha1.PhaseFailed, errors.Wrap(err, "Error getting all staging users")
+	}
+
+	err = r.reconcileStagingUsersConsoleLink(ctx, serverClient, existingStagingUsers, newStagingUsers)
+	if err != nil {
+		r.Log.Error("Error reconciling Staging Console Links", err)
+		return integreatlyv1alpha1.PhaseFailed, errors.Wrap(err, "Error reconciling Staging Console Links")
+	}
+
+	loginEventUserIds, err := r.getUserIdLoginEvents(ctx, serverClient, kc)
+	if err != nil {
+		r.Log.Error("Error getting login events", err)
+		return integreatlyv1alpha1.PhaseFailed, errors.Wrap(err, "Error getting login events")
+	}
+
+	err = r.reconcileTenantRealms(ctx, serverClient, allRealms, mtUsers, loginEventUserIds)
+	if err != nil {
+		r.Log.Error("Error reconcileTenantRealms", err)
+		return integreatlyv1alpha1.PhaseFailed, errors.Wrap(err, "Error reconcileTenantRealms")
+	}
+
+	err = r.reconcileTenantConsoleLinks(ctx, serverClient, mtUsers)
+	if err != nil {
+		r.Log.Error("Error reconciling Tenant Console Links", err)
+		return integreatlyv1alpha1.PhaseFailed, errors.Wrap(err, "Error reconciling Staging Console Links")
+	}
+
+	return integreatlyv1alpha1.PhaseCompleted, nil
+}
+
+func (r *Reconciler) reconcileTenantConsoleLinks(ctx context.Context, serverClient k8sclient.Client, users []userHelper.MultiTenantUser) error {
+	realms, err := r.getAllRealms(ctx, serverClient)
+	if err != nil {
+		r.Log.Error("Error getting all realms", err)
+		return err
+	}
+
+	for _, realm := range realms.Items {
+		if realm.Name == "master" || realm.Name == StagingRealmName {
+			continue
+		}
+		tenantLink := r.Config.GetHost() + "/auth/admin/" + realm.Name + "/console/"
+		if err := r.reconcileDashboardLink(ctx, serverClient, realm.Name, tenantLink, ""); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// If a user logged in to the staging realm but does not have a tenant realm, create one.
+func (r *Reconciler) reconcileTenantRealms(ctx context.Context, serverClient k8sclient.Client, realms *keycloak.KeycloakRealmList, users []userHelper.MultiTenantUser, loginEventUserIds []kccommon.Users) error {
+
+	added, deleted, err := r.getTenantDiff(ctx, serverClient, users, realms.Items, loginEventUserIds)
+	if err != nil {
+		return err
+	}
+
+	for _, user := range added {
+		_, err := r.createTenantRealm(ctx, serverClient, user.TenantName)
+		if err != nil {
+			return err
+		}
+		_, err = r.createTenantKCUser(ctx, serverClient, user)
+		if err != nil {
+			return err
+		}
+		if err = r.removeUserFromStagingRealm(ctx, serverClient, user); err != nil {
+			return err
+		}
+	}
+
+	// remove user from staging realm
```

Comment:
Hey, something small but I think this should be //deleting tenant or so, if I understand correctly we are removing user from staging realm at 119.